RAB: Integrate helper to rest of the tests

ioannad opened the pull request:
Integrates `harness/resizableArrayBufferUtils.js` helper to the rest of the test files in PR #3888.

ptomato commented on test/built-ins/Object/defineProperty/coerced-P-grow.js:
```diff
@@ -0,0 +1,64 @@
+  }
+}
+
+const helper = ObjectDefinePropertyHelper;
```
I'd just name the function `helper` in the first place.

ioannad replied:
Ended up using more descriptive names to match the style of the other RAB tests.

ptomato commented on test/built-ins/Object/defineProperty/coerced-P-shrink.js:
```diff
@@ -0,0 +1,52 @@
+  }
+}
+
+const helper = ObjectDefinePropertyHelper;
```
Same.

ptomato commented on test/built-ins/TypedArray/out-of-bounds-get-and-set.js:
```diff
@@ -0,0 +1,58 @@
+---*/
+
+for (let ctor of ctors) {
+  if (ctor.BYTES_PER_ELEMENT != 1) {
+    continue;
+  }
```
Fine with me if we decide to leave this as is, but is there a reason why we skip non-1-byte typed arrays, instead of multiplying the resizes by `ctor.BYTES_PER_ELEMENT` as in many of the other tests? (I feel like there is probably a good reason, I'm just missing it.)

ioannad replied:
I switched to the style of the other tests and the tests were only failing for BigInt TAs, which was fixed by using `MayNeedBigInt` as in other RAB tests.

ptomato commented on test/built-ins/TypedArray/out-of-bounds-has.js:
```diff
@@ -0,0 +1,44 @@
+---*/
+
+for (let ctor of ctors) {
+  if (ctor.BYTES_PER_ELEMENT != 1) {
+    continue;
+  }
```
Same.

ioannad replied:
Same fix.

ptomato commented on test/built-ins/TypedArray/prototype/slice/coerced-start-grow.js:
Delete this file, it's just an old copy of the one that got renamed to `coerced-start-end-grow.js`.

ioannad replied:
Done

ptomato reviewed (changes requested):
A few things to fix up, then we can finally call this done

ioannad commented:
The latest commit addresses all the review comments, and additionally simplifies superfluous `ReadElement2` and `HasElement2` in `TypedArray/out-of-bounds-behaves-like-detached.js`,

ioannad commented on test/built-ins/Object/defineProperties/typedarray-backed-by-resizable-buffer.js:
```diff
@@ -0,0 +1,225 @@
+  } else {
+    Object.defineProperty(ta, index, { value: value });
+  }
+}
```
Done

Ms2ger commented on test/built-ins/Object/defineProperty/coerced-P-grow.js:
```diff
@@ -11,16 +11,14 @@ includes: [compareArray.js, resizableArrayBufferUtils.js]
 
-function ObjectDefinePropertyHelper(ta, index, value) {
+function MayNeedBigInt(ta, value) {
   if (ta instanceof BigInt64Array || ta instanceof BigUint64Array) {
```
Future work: `MayNeedBigInt` could go into the helper file

ioannad replied:
I'll make a followup PR with this change, since this is defined across many files.

Ms2ger commented on test/built-ins/Object/defineProperty/coerced-P-grow.js:
```diff
@@ -33,7 +31,7 @@ for (let ctor of ctors) {
     }
   };
-  helper(fixedLength, evil, 8);
+  Object.defineProperty(fixedLength, evil, { value: MayNeedBigInt(fixedLength, 8) });
```
Much more readable, thank you

Ms2ger reviewed (approved):
Thanks!
